src/vuln_analysis/server/webhooks/ledger.py

```
@@ -0,0 +1,269 @@
+                "next_retry_at, last_error, dead_lettered_at, lease_owner, lease_expires_at "
+                "FROM webhook_deliveries WHERE job_id = ?",
+                (job_id,),
+            ).fetchone()
+        if row is None:
+            return None
+        return LedgerRecord(**dict(row))
+
+    def list_dead_lettered(self, *, limit: int = 100) -> list[LedgerRecord]:
+        """Return recent dead letters, newest first with job ID as a stable tie-breaker."""
+        if limit < 1:
+            raise ValueError("limit must be at least 1")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)